lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -2442,11 +2448,17 @@ void TypeChecker::diagnoseIfDeprecated(SourceRange ReferenceRange,
   // We match the behavior of clang to not report deprecation warnings
   // inside declarations that are themselves deprecated on all deployment
   // targets.
-  if (Where.isDeprecated()) {
+  if (Where.isDeprecated() || Where.ignoresDeprecation()) {
     return;
   }
 
   auto *ReferenceDC = Where.getDeclContext();
+  auto declImport = ReferenceDC->findImportFor(ReferenceDC->getParentModule());
```

Comment:
The module arguments to both `findImportFor` calls still need changing to the module of the deprecated decl, do you want to change those and add some tests for the use of the attribute on `import` statements, or do you want me to kick off the tests as it currently is?

Reply:
`still need changing to the module of the deprecated decl` However I don't think `Decl` has any properties / methods for getting the parent module, so what should be done here?

Reply:
You can do `DeprecatedDecl->getDeclContext()->getParentModule()`, and same for the extension case

Reply:
Alright, as I fix that, I want to ask a question, how can I create a module to import in the file here? Just create another file and do `@ignoreDeprecationWarnings import file_name`?

Reply:
Take a look at `predates_concurrency_import.swift` for an idea of what to do, the main points are:

- Create a .swift file in the `Inputs` directory of the relevant place you're testing (`test/attr/Inputs` in your case), this will be the module you'll test against
- Do `RUN: %empty-directory(%t)` to create a temporary directory
- Do `%target-swift-frontend -emit-module` to emit the module for the file in the `Inputs` directory, outputting a `.swiftmodule` in the temporary directory you created
- Add `-I %t` to your main test invocation to make the module accessible, allowing it to be imported

Let me know if you have any issues